File: Java/Java source-code/standardize.java


public class standardize {
    public static Double[] standardize_m(Double[] data) {
        double sum = 0;
        double sumSq = 0;
        for (int i = 0; i < data.length; i++) {
            sum += data[i];
            sumSq += data[i] * data[i];
        }
        double mean = sum / data.length;
        double sd = Math.sqrt((sumSq - mean * sum) / data.length);
        for (int i = 0; i < data.length; i++) {
            data[i] = (data[i] - mean) / sd;
        }
        return data;
    }
}

File: Python/standardize.py
import math


def standardize(data):
    suma = 0
    sumSq = 0

    for i in range(0, len(data)):
        suma += data[i]
        sumSq += data[i] * data[i]

    mean = suma / len(data)
    sd = math.sqrt((sumSq - mean * suma) / len(data))

    for i in range(0, len(data)):
        data[i] = (data[i] - mean) / sd

    return data
